packages/common/src/utils/signature_utils.ts

```diff
@@ -7,9 +7,17 @@ import { getStacksVerifier } from '@didtools/pkh-stacks'
 import { getTezosVerifier } from '@didtools/pkh-tezos'
 import { CeramicSigner } from '../ceramic-signer.js'
 import { StreamUtils } from './stream-utils.js'
+import { ServiceMetrics as Metrics } from '@ceramicnetwork/observability'
 
 const DEFAULT_CACAO_REVOCATION_PHASE_OUT_SECS = 24 * 60 * 60
 
+// Metric for when we try to apply a new commit and the CACAO for that commit is expired.
+export const CACAO_EXPIRED_NEW_COMMIT = 'cacao_expired_new_commit'
```

Comment:
It's easy enough to. merge in the dashboard and prometheus pipeline especially. As a rule of thumb, we want metrics generated by the code to be as straightforward and "dumb" as possible (counters, gauges-if-you-must) and more or less all grouping and especially statistical analysis done in monitoring (esp. rates, avgs).

Prometheus history type metrics are an exception because of data volume (historically) and the difficulty of actually doing that in dashboards.

Sounds like in this specific case though, you want a label for the metric, to make aggregation easier:

cacao_expired{reason="existing_state"}
cacao_expired{reason="new_commit"}

That's basically the best of both worlds from a metrics perspective.

Reply:
Good suggestion, updated!